python/cudf/cudf/core/dataframe.py

```diff
@@ -2207,12 +2207,16 @@ def _typecast_before_merge(self, lhs, rhs, left_on, right_on, how):
         def casting_rules(dtype_l, dtype_r, how):
             cast_warn = "can't safely cast column {} from {} with type \
                          {} to {}, upcasting to {}"
-            ctgry_err = "can't implicitly cast column {} to categories \
-                         from right during left join"
+            ctgry_err = "can't implicitly cast column {0} to categories \
+                         from {1} during {1} join"
```

Comment:
No need to change this; just a comment that it might be preferable to do something like:

```python
ctgry_err = ("can't implicitly cast column {column_id} to categories"
                   " from {how} during {how} join")

ctgry_err.format(column_id=rcol, how="right")
```